qiskit/compiler/assemble.py

```diff
@@ -276,13 +283,29 @@ def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
 
     qubit_lo_range = qubit_lo_range or getattr(backend_config, 'qubit_lo_range', None)
     meas_lo_range = meas_lo_range or getattr(backend_config, 'meas_lo_range', None)
-    rep_time = rep_time or getattr(backend_config, 'rep_times', None)
 
-    if isinstance(rep_time, list):
-        rep_time = rep_time[0]
+    dynamic_reprate_enabled = getattr(backend_config, 'dynamic_reprate_enabled', False)
 
+    rep_time = rep_time or getattr(backend_config, 'rep_times', None)
     if rep_time:
-        rep_time = int(rep_time * 1e6)
+        # deprecation warning for ``rep_time`` when dynamic rep rate is enabled
+        if dynamic_reprate_enabled:
+            warnings.warn("Use 'rep_delay' rather than 'rep_time' as dynamic rep rates are enabled "
+                          "on this backend.", DeprecationWarning)
+        if isinstance(rep_time, list):
+            rep_time = rep_time[0]
+        rep_time = rep_time * 1e6 # convert sec to μs
```

Comment:
Why did you remove `int`?

Reply:
Well, to my understanding at least, I believe these are floats?

Reply:
I mean it makes sense to me. Not sure if there was a good reason for it being int before.